Add auto_detect param to convert_values_to_timezone and doc update (#666)
wildlife-dynamics/ecoscope: doc/platform-sdk/content/troubleshooting.md

---

## Developing against a local ecoscope checkout

When developing or debugging ecoscope itself, you can point your workflow at a local checkout instead of the published `ecoscope-platform` conda package. The conda package name is `ecoscope-platform`, but the PyPI/local package name is `ecoscope` with extras:

```yaml
requirements:
  - name: python
    version: "3.12.*"
  - name: rasterio
    version: "1.5.0"
    channel: conda-forge
  - name: ecoscope
    path: /Users/me/ecoscope
    editable: true
    extras: ["platform", "mapping", "analysis"]
```

**Key points:**

- **`python` pin** — without this, the ephemeral discovery environment may
  resolve Python 3.14+, where transitive deps like `pydantic-core` lack
  pre-built wheels.
- **`rasterio` conda dep** — ensures GDAL native libraries are available for
  the editable ecoscope install.
- **`extras`** — all three are required for task discovery to succeed:
    - `platform`: pandera, pydantic, wt-registry, wt-task
    - `mapping`: lonboard, matplotlib (results/map tasks)

>     - `analysis`: statsmodels (analysis tasks)

Reviewer: I think we should expand this comment to mention that depending on what tasks are used, other extras may also be needed? e.g. `plotting`? Not sure if others may ever be needed or that would be it.

- **Post-compile numpy patch** — ecoscope currently pins `numpy>=2,<2.1`,
  which conflicts with what conda resolves on some platforms. After compiling,
  manually edit the generated `pixi.toml` to pin `numpy>=2,<2.1` to match.
  This constraint will be relaxed in a future ecoscope release.
- **Post-compile pydantic patch** — ecoscope requires `pydantic<2.9.0` (newer
  versions break discriminated unions in `apply_classification`), but the
  compiler emits `pydantic>=2.0.0,<3.0.0` as a conda dependency. After
  compiling, manually edit the generated `pixi.toml` to pin

>   `pydantic>=2.0.0,<2.9.0`.

Reviewer: Seems like both of these "Post-compile patches" can be included directly in `requirements:` as conda deps with the appropriate pins, similar to `rasterio`?

Author: Good point. I checked that `numpy` can be added in the requirements, but not pydantic as it will be overriden by the compiler.

- **Separate environment limitation** — `wt-compiler` requires `pydantic>=2.9` while
  `ecoscope[platform]` requires `pydantic<2.9.0`. They cannot coexist in the
  same pixi environment. The compiler does not import ecoscope at runtime — it

>   installs it in an ephemeral subprocess environment for task discovery.

Reviewer: ```suggestion
```
This seems irrelevant? wt-compiler uses ephemeral envs for exactly this reason.

Author: It's a useful note for myself as a habit to put everything together for testing. It might be worth mentioning here to demonstrate the necessity of the ephemeral environment. Maybe drop the word "limitation"?

---

## "Could not read properties of undefined (reading 'name')"

This error appears in Ecoscope Desktop / Ecoscope Web when submitting the configuration form. It means your workflow is missing one or both of the required task instances: `workflow_details` and `time_range`.

Every ecoscope workflow **must** include both:

> ```yaml
> workflow:
>   - name: Workflow Details
>     id: workflow_details
>     task: set_workflow_details
> 
>   - name: Time Range
>     id: time_range
>     task: set_time_range
>     partial:
>       time_format: '%d %b %Y %H:%M:%S'
> ```
> 

Reviewer: ````suggestion
## "Could not read properties of undefined (reading 'name')"

This error appears in Ecoscope Desktop / Ecoscope Web when submitting the configuration form. It means your workflow is missing the required task instance `workflow_details`.

Every ecoscope workflow **must** include:

```yaml
workflow:
  - name: Workflow Details
    id: workflow_details
    task: set_workflow_details
```

````

Reviewer: Time range is not required anymore, right? I think that was as a change Ray made in response to PGAFF workflows not always needing a time range. Also, error for a malformed time range would be different ("name" is not a field on time range)... my suggestion is we make this section exclusively about workflow_details.

> **Fix**: Add the missing task(s) to your `spec.yaml`. The `id` values must be exactly `workflow_details` and `time_range` — Ecoscope Desktop and Ecoscope Web look for these specific IDs when processing the configuration form.

Reviewer: ```suggestion
**Fix**: Add the missing task to your `spec.yaml`. The `id` values must be exactly `workflow_details` — Ecoscope Desktop and Ecoscope Web look for this specific ID when processing the configuration form.
```

    - `analysis`: statsmodels (analysis tasks)
    - `plotting`: plotly, scikit-learn

    Other extras may also be needed depending on your tasks/dependencies.
- **`numpy` pin** — ecoscope currently pins `numpy>=2,<2.1`, which conflicts
  with what conda resolves on some platforms. Add a numpy conda requirement to
  the spec to avoid a post-compile patch:

    ```yaml
    - name: numpy
      version: ">=2,<2.1"
      channel: conda-forge
    ```

- **Post-compile pydantic patch** — ecoscope requires `pydantic<2.9.0` (newer
  versions break discriminated unions in `apply_classification`), but the
  compiler emits `pydantic>=2.0.0,<3.0.0` as a conda dependency. After
  compiling, manually edit the generated `pixi.toml` to pin
  `pydantic>=2.0.0,<2.9.0`. This cannot be set via `requirements:` because
  the compiler overrides pydantic's version constraint.
- **Separate environments** — `wt-compiler` requires `pydantic>=2.9` while
  `ecoscope[platform]` requires `pydantic<2.9.0`. They cannot coexist in the
  same pixi environment, which is why the compiler installs requirements in an
  ephemeral subprocess environment for task discovery rather than importing

>   them directly.

Reviewer: ```suggestion
```

IMHO this bullet does not add any actionable/useful information to the topic being covered in this section. This section is already quite advanced/intricate, so I would strongly favor not including any extraneous information. Also, this bullet is unnecessarily specific: wt-compiler does not use an ephemeral environment for pydantic incompatibility specifically, but as as hedge against version conflicts more generally (of which pydantic is just one of possibly 100s or 1000s of such cases). I would favor removing this bullet.
